Diplom_2 - Devel

K0tani opened the pull request.

K0tani committed:
add code

K0tani committed:
add allure report

pomegranate282922 commented on src/main/java/praktikum/Ingredients.java:
```diff
@@ -0,0 +1,7 @@
+
+@Data
+public class Ingredients {
+    public String _id;
```
Нарушаешь принцип инкапсуляции

K0tani replied:
вместо public поставил private в Ingredients.java

pomegranate282922 commented on src/main/java/praktikum/IngredientsRequest.java:
```diff
@@ -0,0 +1,10 @@
+@AllArgsConstructor
+public class IngredientsRequest {
+
+    List<String> ingredients;
```
Модификатор доступа поставь минимальный из возможных

K0tani replied:
- добавил модификатор доступа в IngredientsRequest.java

pomegranate282922 commented on src/main/java/praktikum/OrderClient.java:
```diff
@@ -0,0 +1,49 @@
+                .header("Authorization", token)
+                .body(ingredientsRequest)
+                .when()
+                .post(PATH)
```
Ты же указал base uri в req spec. Зачем опять оно тут?

K0tani replied:
- бес попутал (или заучился\заработался), убрал url который уже есть OrderClient.java

pomegranate282922 commented on src/main/java/praktikum/UserGenerator.java:
```diff
@@ -0,0 +1,22 @@
+@AllArgsConstructor
+public class UserGenerator {
+
+    public String email;
```
Везде исправь модификаторы доступа согласно принципу инкапсуляции

K0tani replied:
- заменил доступы с public на private UserGenerator.java

pomegranate282922 commented on src/test/java/praktikum/user/UserRegistrationWithEmptyFieldsTest.java:
```diff
@@ -0,0 +1,60 @@
+                .extract()
+                .path("success");
+
+    }
```
Здесь ты надеешься на зелёный тест и не удаляешь юзера, если он создастся в результате пойманного бага. Предусмотри такую ситуацию тоже.

K0tani replied:
- добавил clear пользователя, если он был случайно создан UserRegistrationWithEmptyFieldsTest.java

pomegranate282922 reviewed (commented):
Отлично. Есть пара замечаний - поправь их, пожалуйста.

K0tani committed:
- вместо public поставил private в Ingredients.java
- добавил модификатор доступа в IngredientsRequest.java
- бес попутал (или заучился\заработался), убрал url который уже есть OrderClient.java
- заменил доступы с public на private UserGenerator.java
- добавил clear пользователя, если он был случайно создан UserRegistrationWithEmptyFieldsTest.java

pomegranate282922 commented on src/main/java/praktikum/IngredientsResponse.java:
```diff
@@ -0,0 +1,8 @@
+@Getter
+public class IngredientsResponse {
+
+    public Ingredients[] data;
```
Оу.. а тут?!

K0tani replied:
Не внимательным я конечно бываю

pomegranate282922 replied:
Тыжтестировщик)

pomegranate282922 commented on src/main/java/praktikum/IngredientsClient.java:
```diff
@@ -0,0 +1,15 @@
+import io.restassured.response.Response;
+import static io.restassured.RestAssured.given;
+public class IngredientsClient extends RestAssuredClient {
+    public final String PATH = "https://stellarburgers.nomoreparties.site/api/" + "ingredients";
```
И здесь не нужен полный путь

pomegranate282922 reviewed (approved):
Зачтено!

K0tani committed:
fix
